File: 2020-x-sentimental-cash/cash.py
from cs50 import get_float
from sys import exit


while True:
    change = get_float("Change owed: ")
    if change < 0:
        print("Repeat")
    else:
        i = 0
        no_coins = change // 0.25
        change = change % 0.25
        change = round(change, 2)
        no_coins += change // 0.10
        change = change % 0.10
        change = round(change, 2)
        no_coins += change // 0.05
        change = change % 0.05
        change = round(change, 2)
        no_coins += change // 0.01
        print(f"{round(no_coins)}")
        exit(0)

File: 2019-x-cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float owed;
    int owed_i;
    do
    {
        owed = get_float("Change owed: ");
        owed_i = round(owed*100);
    }
    while (owed_i < 0);
    int st_25 = round(owed_i/25);
    int ost1 = owed_i % 25;
    int st_10 = round(ost1/10);
    int ost2 = ost1 % 10;
    int st_5 = round(ost2/5);
    int ost3 = ost2 % 5;
    int st_1 = round(ost3/1);
    printf("%i\n", st_1 + st_5 + st_10 + st_25);
    
}
